### PR TITLE
Add API for multi-dc Sharding/Singleton in Typed, #27705

### PR DESCRIPTION
**Pull request by patriknw**
* add withDataCenter in Entity, following same patterna as the role
* update cluster-dc.md, split in classic and new pages

Refs #27705

I will add what is missing for Singleton in this PR, but reviews can begin.

### CODE REVIEW

File: akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/javadsl/ClusterSharding.scala [hunk 1]
```diff
@@ -187,6 +188,18 @@ abstract class ClusterSharding {
    */
   def entityRefFor[M](typeKey: EntityTypeKey[M], entityId: String): EntityRef[M]
 
+  /**
+   * Create an `ActorRef`-like reference to a specific sharded entity running in another data center.
+   *
+   * You have to correctly specify the type of messages the target can handle via the `typeKey`.
+   *
+   * Messages sent through this [[EntityRef]] will be wrapped in a [[ShardingEnvelope]] including the
+   * here provided `entityId`.
```
Comment (helena) [inline akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/javadsl/ClusterSharding.scala]:
```suggestion
   * provided `entityId`.
```

File: akka-docs/src/main/paradox/cluster-dc.md [hunk 2]
```diff
@@ -103,58 +15,10 @@ Scala
...
-The reason for only using a limited number of nodes is to keep the number of connections across data
-centers low. The same nodes are also used for the gossip protocol when disseminating the membership
-information across data centers. Within a data center all nodes are involved in gossip and failure detection.
-
-This influences how rolling upgrades should be performed. Don't stop all of the oldest that are used for gossip
-at the same time. Stop one or a few at a time so that new nodes can take over the responsibility.
-It's best to leave the oldest nodes until last.
-
-See the @ref:[failure detector](typed/cluster.md#failure-detector) for more details.
+For the full documentation of this feature and for new projects see @ref:[Multi-DC Cluster](typed/cluster-dc.md#membership).
```
Comment (helena) [inline akka-docs/src/main/paradox/cluster-dc.md]: 👍

**Review by helena — APPROVED** [review]
Looks great, I did just a cursory check on the java code however.

File: akka-cluster-typed/src/test/java/jdocs/akka/cluster/typed/SingletonCompileOnlyTest.java [hunk 3]
```diff
@@ -117,4 +117,12 @@ public static void backoff() {
     // #backoff
     proxy.tell(Counter.Increment.INSTANCE); // avoid unused warning
   }
+
+  public static void dcProxy() {
+    // #create-singleton-proxy-dc
+    ActorRef<Counter.Command> singletonProxy =
+        ClusterSingleton.get(system).init(SingletonActor.of(Counter.create(), "GlobalCounter"));
```
Comment (chbatey) [inline akka-cluster-typed/src/test/java/jdocs/akka/cluster/typed/SingletonCompileOnlyTest.java]: missing with dc setting

File: akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/scaladsl/ClusterSharding.scala [hunk 4]
```diff
@@ -187,6 +189,18 @@ trait ClusterSharding extends Extension { javadslSelf: javadsl.ClusterSharding =
...
+   * Create an `ActorRef`-like reference to a specific sharded entity running in another data center.
+   *
+   * You have to correctly specify the type of messages the target can handle via the `typeKey`.
+   *
+   * Messages sent through this [[EntityRef]] will be wrapped in a [[ShardingEnvelope]] including the
+   * here provided `entityId`.
+   *
+   * For in-depth documentation of its semantics, see [[EntityRef]].
+   */
+  def entityRefFor[M](typeKey: EntityTypeKey[M], entityId: String, dataCenter: DataCenter): EntityRef[M]
```
Comment (chbatey) [inline akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/scaladsl/ClusterSharding.scala]: :+1:

**Review by chbatey — APPROVED** [review]
LGTM with one comment

File: akka-cluster-typed/src/test/java/jdocs/akka/cluster/typed/SingletonCompileOnlyTest.java (hunk 3 above)
Comment (patriknw) [reply akka-cluster-typed/src/test/java/jdocs/akka/cluster/typed/SingletonCompileOnlyTest.java]: oh, thanks

**Review by johanandren — APPROVED** [review]
LGTM

**Pull request merged**